### DISCUSSION TITLE
PHDR segment not covered by LOAD segment

### DISCUSSION THREAD

**Discussion by netITea** [open-watcom/open-watcom-v2#802] [General]
Hello,
I try to compile ow with yocto on linux x86_64 and in the final build phase appears this error message. In this phase yocto runs "strip --remove-section=.comment --remove-section=.note" on binaries. And some of them (wcc, wccaxp, wd, wasppc, wcc386, wlink, wccppc, wcl, ...) from the binl directory have this problem.
Any idea about this issue?
Andrey

**Comment by jmalak** [comment]
I don't known what yocto is.
Question is why "strip --remove-section=.comment --remove-section=.note" is run and what it to do with executables.
Anyway for OW build strip command has no sense and can be source of problem because the OW executable use some additional data after executable size and if your implementation of strip doesn't remove only defined sections but remove other data then it damage it.

**Reply by netITea** [reply]
Yocto/Openembedded (https://www.yoctoproject.org) is a build system for embedded linux distributions. And binary stripping seems to be a default procedure for sysroot_staging-task.
Ok, I'll try to disable binary striping.
Thanks for your very fast response. Regards Andrey

**Reply by jmalak** [reply]
If it is standard task then it needs to be done before OW wstrip command which add extra information to executable.
It is last command which works with final executable.
Anyway could you give me OW build log which show the problem.
What toolchain you use for OW build?

**Reply by netITea** [reply]
Hi,
I put some logs from building tasks to an archive together.
You may need some input for better understanding of the build process.
do_patch applies a patch on setvars.sh to enable dosbox
do_configure executes build.sh boot
do_compile executes build.sh build
do_install executes build.sh cprel and copies content of 
the rel-directory to /opt/ow
And in the Logs for do_populate_sysroot you can see the problem I mentioned.
Strip-skipping seams to solve my problem.
[logs.zip](https://github.com/open-watcom/open-watcom-v2/files/7887837/logs.zip)

**Reply by jmalak** [reply]
I will look on this. 
If strip is standard for Yocto then we can modify OW build configuration to properly work in Yocto environment to move wstrip command after complete executable build by Yocto (including strip command). But if strip command is used after OW build then we can not fix it in OW build process.

**Reply by jmalak** [reply]
I think I understand the process.
It looks like strip command is used to minimize size of executable by removing unnecessary sections before sys filesystem is prepared.
Try to look on strip command help if it is possible to add some option for strip command which ensure that data after standard ELF executable image will be safe (not removed from image).
Anyway the message could be related to ELF header problem, but OW doesn't use anything specific, it relies on host linker with minimal command line. If Yocto requires some special linker options we can add it to OW build make files.

**Reply by jmalak** [reply]
I tested it on my Linux test box by command
strip --remove-section=.comment --remove-section=.note wccaxp
and get the same error

strip: stC34TK5: error: PHDR segment not covered by LOAD segment
strip:stC34TK5[.text]: Neplatná operace

It is issue with strip command and ELF executable generated by OW linker.
I raised issue #803 to be fixed.

**Reply by jmalak** [reply]
it is fixed in git repository by commit f09e0c969c45679c048180f2dc6b3dbbe69e42a0

**Reply by jmalak** [reply]
second fix for ELF executable image which fix order of symbols in symbol table was submitted to git repository.
Now you should not have a problem with stripping info from OW ELF executable.

**Reply by netITea** [reply]
Thank you!
